asgiref/sync.py

```diff
@@ -269,7 +281,13 @@ def __get__(self, parent, objtype):
         return functools.update_wrapper(func, self.awaitable)
 
     async def main_wrap(
-        self, args, kwargs, call_result, source_thread, exc_info, context
+        self,
+        args,
+        kwargs,
```

Comment:
you need to type annotate all the parameters for mypy to check this function

It's probably worth making all the parameters positional only so you can use `*args` and `**kwargs`:


```python
def _main_wrap(call_result: Future[R], source_thread: threading.Thread, exc_info: ExcInfo, context: Optional[Context]) -> Callable[P, Coroutine[Any, Any, None]:
    async def wrap(*args: P.args, **kwargs: P.kwargs) -> None:
        ...

    return wrap
```